ci.sh

```diff
@@ -37,25 +37,43 @@ python -c "import sys, struct, ssl; print('python:', sys.version); print('versio
 echo "::endgroup::"
 
 echo "::group::Install dependencies"
-python -m pip install -U pip build
+python -m pip install -U pip uv build
 python -m pip --version
+uv --version
+uv venv .venv --seed
+
+if [ $(python -c "import sys;print(sys.platform.startswith('win'))") = "True" ]; then
+    source ./.venv/Scripts/activate
+else
+    source .venv/bin/activate
+fi
+
+# Make sure pip is installed (pypy weirdness)
+python -m ensurepip
+
+# Install uv in venv
+python -m pip install --upgrade uv
+
+# Make sure build is there
+uv pip install build
```

Comment:
Why do we need to create a virtualenv? In CI it shouldn't be necessary, and for developers I feel like it's better to let the developer be responsible for handling their venv and not recreate the venv every time you run `ci.sh`

Reply:
I think there were some issues because I didn't know about `--system` mode. Otherwise, on default, `uv` uses a virtual environment instead of touching system python installation. Thanks for reminding me about this, I'll get that fixed.

Reply:
I believe I have fixed this